core/src/main/scala/akka/kafka/internal/SingleSourceLogic.scala

```diff
@@ -51,7 +52,14 @@ import scala.concurrent.{Future, Promise}
     if (!isClosed(shape.out)) {
       complete(shape.out)
     }
-    sourceActor.become(shuttingDownReceive)
+    sourceActor.become(shuttingDownReceive.orElse {
+      case (_, Messages(requestId, messages)) =>
+        // Prevent stage failure during shutdown by ignoring Messages
+        if (messages.hasNext)
+          log.debug("Unexpected message `Messages` received with requestId={} and a non-empty message iterator: {}",
```

Comment:
Reads weird 'message `Messages`'. We can maybe drop 'message'.

```suggestion
          log.debug("Unexpected `Messages` received with requestId={} and a non-empty message iterator: {}",
```

I didn't understand why this is related with the changes for telemetry though. Have run into tests errors that required this?

Reply:
About my question above, please ignore it, you mentioned it in the description.

> fixed MatchError resulting in the stage failure

Reply:
I like to use the `Hide` feature to reduce noise. :-) 

<img width="341" alt="Screenshot 2021-07-23 at 08 52 34" src="https://user-images.githubusercontent.com/762126/126746980-ad016292-e6ae-4c23-b403-586d102b642f.png">

Reply:
This is not off-topic. There is a code suggestion.

Reply:
Apologies, I don't know what the hell happened there. I think I read a PR with comments of yours where you added corrections to yourself and I used the `Hide` feature on the wrong PR.